src/generator.cpp

```diff
@@ -1868,7 +1868,7 @@ double processEquationMultiplierAst(const GeneratorEquationAstPtr &ast, std::vec
 
             // The only time we check multiplier mismatch is in a comparision operation.
             if (isDirectComparisonOperator(ast)) {
-                if (leftMult != rightMult && ast->mLeft != nullptr && ast->mRight != nullptr) {
+                if ((leftMult > rightMult || leftMult < rightMult) && ast->mLeft != nullptr && ast->mRight != nullptr) {
```

Comment:
Wow, this is *ugly*! Please use [`areEqual()`](https://github.com/cellml/libcellml/blob/4f19a5c917c60f7f345ca11d9f7649d33d8b3bac/src/utilities.cpp#L163-L166) instead.